File: app/src/main/java/com/kun/dagger2rxjava2retrofit2demo/activities/main/MainPresenter.java
package com.kun.dagger2rxjava2retrofit2demo.activities.main;

import com.kun.baselib.base.BasePresenter;
import com.kun.baselib.base.BaseResponse;
import com.kun.baselib.net.HttpSubscriber;
import com.kun.baselib.utils.RxHelper;
import com.kun.dagger2rxjava2retrofit2demo.bean.WeatherResponse;
import com.kun.dagger2rxjava2retrofit2demo.net.AppNetApi;

import javax.inject.Inject;

/**
 * @author kun
 * @date 2017/10/24
 */

public class MainPresenter extends BasePresenter<MainContract.View> implements MainContract.Present{
    @Inject
    MainPresenter() {
    }
    @Inject
    AppNetApi mApi;
    @Override
    public void getWeather(String city) {
        mApi.getWeather(city)
                .compose(mView.getLifecycleProvider().<BaseResponse<WeatherResponse>>bindToLifecycle())
                .compose(RxHelper.<BaseResponse<WeatherResponse>>io_main())
                .subscribe(new HttpSubscriber<BaseResponse<WeatherResponse>>() {
                    @Override
                    public void onSuccess(BaseResponse<WeatherResponse> weatherResponseBaseResponse) {
                        if (mView != null) {
                            mView.getWeatherSuccess(weatherResponseBaseResponse.getData());
                        }
                    }
                });
    }
}

File: app/src/main/java/com/kun/dagger2rxjava2retrofit2demo/activities/main/MainPresenter.kt
package com.kun.dagger2rxjava2retrofit2demo.activities.main

import com.kun.baselib.base.BaseDataCache
import com.kun.baselib.base.BaseResponse
import com.kun.baselib.net.RxSubscriber
import com.kun.baselib.utils.RxHelper
import com.kun.dagger2rxjava2retrofit2demo.base.BasePresent
import com.kun.dagger2rxjava2retrofit2demo.bean.WeatherResponse
import com.kun.dagger2rxjava2retrofit2demo.net.AppNetApi
import com.trello.rxlifecycle2.LifecycleProvider
import com.trello.rxlifecycle2.android.ActivityEvent

import javax.inject.Inject

/**
 * Created by kun on 2017/8/14.
 */

class MainPresenter @Inject
internal constructor(view: MainContract.View, lifecycleProvider: LifecycleProvider<ActivityEvent>, api: AppNetApi, mDataCache: BaseDataCache) : BasePresent<MainContract.View, ActivityEvent>(view, lifecycleProvider, api, mDataCache), MainContract.Present {
    @Inject
    internal fun setupListener() {
        mView.setPresenter(this)
    }

    override fun getWeather(city: String) {
        mApi.getWeather(city).compose(lifecycleProvider.bindToLifecycle())
                .compose(RxHelper.io_main())
                .subscribe(object : RxSubscriber<BaseResponse<WeatherResponse>>() {
                    override fun onSuccess(weatherResponseBaseResponse: BaseResponse<WeatherResponse>) {
                        mView.getWeatherSuccess(weatherResponseBaseResponse.data)
                    }
                })
    }
}
